prover/src/universal_batch_verifier/compute_submission_id.rs

```diff
@@ -0,0 +1,54 @@
+use circuits::{
+    batch_verify::universal::native::{
+        compute_circuit_id, json::load_app_vk_proof_and_inputs_batch,
+    },
+    keccak::utils,
+};
+use clap::Parser;
+use core::iter;
+
+#[derive(Parser, Debug)]
+pub struct ComputeSubmissionIdParams {
+    /// JSON file containing a batch of app_vk, proof, public input triples
+    /// to be verified by the BatchVerifier circuit.
+    #[arg(short = 'b', long, value_name = "app-vk-proof-batch-file")]
+    pub(crate) app_vk_proof_batch: String,
+
+    /// Num proof ids
+    #[arg(short = 'n', long, value_name = "num_proof_ids")]
+    pub(crate) num_proof_ids: Option<usize>,
+
+    /// Submission size
+    #[arg(short = 's', long, value_name = "submission-size")]
+    pub(crate) submission_size: Option<usize>,
```

Comment:
What's the thinking behind these optional args? Is the idea that the app_vk_proof file may contain more proofs than we want to include in this submission? 

I think it would probably be fine to just remove the optional args and always just let `num_proof_ids` be the length of the JSON file and let submission size be `num_proof_ids` rounded up to a power of two. But maybe I'm not seeing the use case you have in mind

Reply:
AFAIK the `upa compute` equivalent does not have the num-proof-ids option.